internal/core/thirdparty/tantivy/tantivy-binding/src/array.rs

```diff
@@ -0,0 +1,37 @@
+use core::slice;
+
+use libc::{size_t, freeaddrinfo};
+use scopeguard::defer_on_unwind;
+
+#[repr(C)]
+pub struct RustArray {
+    array: *mut u32,
+    len: size_t,
+}
+
+impl RustArray {
+    pub fn from_vec(vec: Vec<u32>) -> RustArray {
+        defer_on_unwind!({
+            ::std::process::abort();
+        });
+        let v: Box<[u32]> = vec.into_boxed_slice();
+        RustArray {
+            len: v.len().try_into().expect("Integer Overflow"),
```

Comment:
Vec has capacity properties. use len as here, may cause bug in some memory allocator.

Reply:
fixed.